### PR TITLE
Added all private endpoint APIs

### PR DESCRIPTION
**Pull request by LarryOsterman**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

This pull request adds PrivateLink required APIs to the MAA Service. All of these are required by privatelink and are not intended for customers to call - we're adding them to our swagger because they are needed for non-customer facing scenarios. As such, they should not be considered breaking changes.

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [X] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [] SDK of .NET (need service team to ensure code readiness)
      - [X] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [X] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [X] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [X] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [X] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [X] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [X] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [X] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by LarryOsterman** [comment]
New API version is not required, the GA API surface has not yet been made available to customers. Also the newly added APIs are not directly customer facing, they are for PrivateLink support.

**Comment by j5lim** [comment]
> New API version is not required, the GA API surface has not yet been made available to customers. Also the newly added APIs are not directly customer facing, they are for PrivateLink support.

Please follow the action described in Breaking Change Review Checklist and get an exception if you think this does not break any customers.

**Review by j5lim — APPROVED** [review]
Breaking changes were reviewed by Azure Breaking Changes reviewers and approved.

File: specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json [hunk 1]
```diff
@@ -591,29 +599,224 @@
...
-      "description": "Client supplied parameters used to create a new attestation service instance.",
+      "description": "Client supplied parameters used to create a new attestation provider.",
       "properties": {
         "policySigningCertificates": {
           "description": "JSON Web Key Set defining a set of X.509 Certificates that will represent the parent certificate for the signing certificate used for policy operations",
           "$ref": "../../../../../common-types/rfcs/rfc7517.json#/definitions/JSONWebKeySet"
         }
       }
+    },
+    "PrivateEndpointConnection": {
```
Comment (akning-ms) [inline specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json:626]: Can we reuse common definition in "../../../../../common-types/resource-management/v1/privatelinks.json"?
If your service implementation is match with common definition

**Comment by akning-ms** [comment]
Can you change subject to align real change? looks it is not about dataplane api any more. mainly for private endpoint API

File: specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json (hunk 1 above)
Comment (LarryOsterman) [reply specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json:626]: 7 lint warnings? I see the Java SDK error but that's the only one I'm seeing.

**Comment by LarryOsterman** [comment]
> Can you change subject to align real change? looks it is not about dataplane api any more. mainly for private endpoint API

Subject is updated.

File: specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json [hunk 2]
```diff
@@ -591,29 +599,166 @@
...
       "properties": {
         "policySigningCertificates": {
           "description": "JSON Web Key Set defining a set of X.509 Certificates that will represent the parent certificate for the signing certificate used for policy operations",
           "$ref": "../../../../../common-types/rfcs/rfc7517.json#/definitions/JSONWebKeySet"
+        },
+        "privateEndpointConnections": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PrivateEndpointConnectionItem"
```
Comment (akning-ms) [inline specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json:628]: We can just ref to here, here is root entry.  rest of definition is not required if so. right?
 "$ref": "../../../../../common-types/resource-management/v1/privatelinks.json#/definitions/PrivateEndpointConnection"

File: specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json (hunk 1 above)
Comment (akning-ms) [reply specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json]: ⚠️ R2001 - AvoidNestedProperties | Consider using x-ms-client-flatten to provide a better end user experienceNew: Microsoft.Attestation/stable/2020-10-01/attestation.json#L678
-- | --
⚠️ R4021 - DescriptionAndTitleMissing | 'PrivateLinkConnectionItemProperties' model/property lacks 'description' and 'title' property. Consider adding a 'description'/'title' element. Accurate description/title is essential for maintaining reference documentation.New: Microsoft.Attestation/stable/2020-10-01/attestation.json#L643
⚠️ R4021 - DescriptionAndTitleMissing | 'provisioningState' model/property lacks 'description' and 'title' property. Consider adding a 'description'/'title' element. Accurate description/title is essential for maintaining reference documentation.New: Microsoft.Attestation/stable/2020-10-01/attestation.json#L646
⚠️ R4021 - DescriptionAndTitleMissing | 'privateEndpoint' model/property lacks 'description' and 'title' property. Consider adding a 'description'/'title' element. Accurate description/title is essential for maintaining reference documentation.New: Microsoft.Attestation/stable/2020-10-01/attestation.json#L649
⚠️ R4021 - DescriptionAndTitleMissing | 'privateLinkServiceConnectionState' model/property lacks 'description' and 'title' property. Consider adding a 'description'/'title' element. Accurate description/title is essential for maintaining reference documentation.New: Microsoft.Attestation/stable/2020-10-01/attestation.json#L658
⚠️ R4021 - DescriptionAndTitleMissing | 'properties' model/property lacks 'description' and 'title' property. Consider adding a 'description'/'title' element. Accurate description/title is essential for maintaining reference documentation.New: Microsoft.Attestation/stable/2020-10-01/attestation.json#L678

File: specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json [hunk 3]
```diff
@@ -525,6 +525,14 @@
         "attestUri": {
           "type": "string",
           "description": "Gets the uri of attestation service"
+        },
+        "privateEndpointConnections": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "../../../../../common-types/resource-management/v1/privatelinks.json#/definitions/PrivateLinkServiceConnectionState"
```
Comment (akning-ms) [inline specification/attestation/resource-manager/Microsoft.Attestation/stable/2020-10-01/attestation.json:533]: Per my comment in last time, here should be PrivateEndpointConnection,  not PrivateLinkServiceConnectionState. right?

**Pull request merged**